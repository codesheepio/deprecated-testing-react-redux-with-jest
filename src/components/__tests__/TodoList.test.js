import React from 'react'
import { shallow } from 'enzyme' // eslint-disable-line import/no-extraneous-dependencies
import toJson from 'enzyme-to-json' // eslint-disable-line import/no-extraneous-dependencies
import TodoList from '../TodoList'
import TodoItem from '../TodoItem'

describe('TodoList', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      todos: [
        { id: 1, text: 'Feed cat', complete: false },
        { id: 2, text: 'Drink coffee', complete: true },
      ],
    }
    wrapper = shallow(<TodoList {...props} />)
  })

  it('matches its snapshot - empty', () => {
    wrapper.setProps({ todos: [] })

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches its snapshot - with todos', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct structure', () => {
    expect(wrapper.is('ul')).toBe(true)
    expect(wrapper.hasClass('todo-list')).toBe(true)
  })

  it('renders todo items from props', () => {
    expect(wrapper.find(TodoItem).length).toBe(props.todos.length)
  })
})
