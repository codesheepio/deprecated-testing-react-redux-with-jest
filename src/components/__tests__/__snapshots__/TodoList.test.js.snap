// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches its snapshot - empty 1`] = `
<ul
  className="todo-list"
/>
`;

exports[`TodoList matches its snapshot - with todos 1`] = `
<ul
  className="todo-list"
>
  <TodoItem
    complete={false}
    completed={false}
    id={1}
    text="Feed cat"
  />
  <TodoItem
    complete={true}
    completed={false}
    id={2}
    text="Drink coffee"
  />
</ul>
`;
